### PR TITLE
[CONTINT-4412] Upgrade `k8s` dependencies

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Update kubernetes and related dependencies.

### Motivation

We need to update `datadog-operator` to be able to [update in `datadog-agent`](https://github.com/DataDog/datadog-agent/blob/6d5f968e0dd5ee4a7504d780e398f876d5c8de27/go.mod#L157).

### Additional Notes

`PodSecurityPolicy` has been dropped upstream by https://github.com/kubernetes/api/commit/5a75c6e751ef86e3cc1bb758c2fc8c30df53b03c.

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### Checklist

- [X] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [X] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,52 +1,54 @@
...
 
-go 1.22
+go 1.22.0
 
 toolchain go1.22.7
 
 require (
 	github.com/DataDog/datadog-api-client-go/v2 v2.27.0
-	github.com/DataDog/extendeddaemonset v0.10.0-rc.4
+	github.com/DataDog/extendeddaemonset v0.9.0-rc.3.0.20240930151437-dfe15f763532
```
Comment (L3n41c) [inline go.mod:9]: To be updated once DataDog/extendeddaemonset#192 is merged.

**Review by michaelcretzman — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -1,86 +1,96 @@
...
 
-go 1.22
+go 1.22.0
 
 toolchain go1.22.7
 
 require (
 	github.com/DataDog/datadog-api-client-go/v2 v2.27.0
-	github.com/DataDog/extendeddaemonset v0.10.0-rc.4
+	github.com/DataDog/extendeddaemonset v0.9.0-rc.3.0.20241010101357-2f0cfa774277
```
Comment (L3n41c) [inline go.mod:9]: This is the commit of DataDog/extendeddaemonset#192.

To be updated as soon as this PR is merged.

File: api/k8s.io/policy/v1beta1/register.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+// Copyright 2024-present Datadog, Inc.
+
+package v1beta1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
```
[the remark this answers is not in the document]
Comment (L3n41c) [reply api/k8s.io/policy/v1beta1/register.go]: PSP references have been removed.

**Review by levan-m — APPROVED** [review]

**Pull request merged**